File: 4 - lottery/python/parsed_input.py
parsed_input = [];

with open("../input.txt") as file:
    for line in file:
        [card_part, numbers_part] = line.split(": ")
        number = int(card_part.split()[1])
        [winning_numbers, your_numbers] = numbers_part.split(" | ")
        winning_numbers = [int(x) for x in winning_numbers.split()]
        your_numbers = [int(x) for x in your_numbers.split()]
        parsed_input.append({
            "winning_numbers": winning_numbers,
            "your_numbers": your_numbers,
            "number": number
        })

File: 7 - house of cards/rust/src/parsed_input.rs
use std::fs::File;
use std::io::{BufReader, BufRead};

use crate::game::Game;
use crate::replace_jokers::replace_jokers;

fn cards_to_numbers(s: &str) -> Vec<i32> {
    s.chars()
        .map(|x: char| if x.is_numeric() {x.to_digit(10).unwrap() as i32} else {match x {
            'A' => 14,
            'K' => 13,
            'Q' => 12,
            'J' => 11,
            'T' => 10,
            _ => panic!("Unknown character {} encountered.", x)
        }})
        .collect()
}

pub fn parse_input(do_replace_jokers: bool) -> Vec<Game> {
    let file = File::open("../input.txt").unwrap();
    let lines = BufReader::new(file).lines();
    let mut games: Vec<Game> = Vec::new();

    for line in lines {
        let string = line.unwrap();
        let mut parts = string.split(" ");
        let cards = cards_to_numbers(parts.next().unwrap());
        let bid: i32 = parts.next().unwrap().parse().unwrap();
        let game = if do_replace_jokers {replace_jokers(&cards, bid)} else {Game::new(cards, bid)};
        games.push(game)
    }

    games
}